From 6a5639646aae6bc9a0f7a78f52d62ad7b6b3fffb Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 22 May 2013 15:14:09 +0300
Subject: ARM: i.MX6: cm-fx6: uart: add uart1 board support

Add uart1 board support.
Configure:
	UART1_TXD function on the SD3_DAT7 pin
	UART1_RXD function on the SD3_DAT6 pin
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |    3 +++
 arch/arm/mach-mx6/board-cm-fx6-q.h  |    4 ++++
 arch/arm/mach-mx6/board-cm-fx6.c    |    1 +
 3 files changed, 8 insertions(+)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index 44e69d8..eb902e6 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -19,6 +19,9 @@
 #include <mach/iomux-mx6dl.h>
 
 static iomux_v3_cfg_t cm_fx6_dl_pads[] = {
+	/* UART1 */
+	MX6DL_PAD_SD3_DAT7__UART1_TXD,
+	MX6DL_PAD_SD3_DAT6__UART1_RXD,
 	/* UART4 for debug */
 	MX6DL_PAD_KEY_COL0__UART4_TXD,
 	MX6DL_PAD_KEY_ROW0__UART4_RXD,
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index 160ad21..4470ada 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -19,6 +19,10 @@
 #include <mach/iomux-mx6q.h>
 
 static iomux_v3_cfg_t cm_fx6_q_common_pads[] = {
+	/* UART1 */
+	MX6Q_PAD_SD3_DAT7__UART1_TXD,
+	MX6Q_PAD_SD3_DAT6__UART1_RXD,
+
 	/* UART4 for debug */
 	MX6Q_PAD_KEY_COL0__UART4_TXD,
 	MX6Q_PAD_KEY_ROW0__UART4_RXD,
diff --git a/arch/arm/mach-mx6/board-cm-fx6.c b/arch/arm/mach-mx6/board-cm-fx6.c
index 7f9a357..7514acc 100644
--- a/arch/arm/mach-mx6/board-cm-fx6.c
+++ b/arch/arm/mach-mx6/board-cm-fx6.c
@@ -290,6 +290,7 @@ static const struct imxuart_platform_data cm_fx6_uart1_data __initconst = {
 
 static inline void cm_fx6_init_uart(void)
 {
+	imx6q_add_imx_uart(0, NULL);
 	imx6q_add_imx_uart(3, NULL);
 	imx6q_add_imx_uart(1, &cm_fx6_uart1_data);
 	imx6q_add_imx_uart(4, NULL);
-- 
1.7.9.5

